src/search/search.go

```diff
@@ -66,72 +52,92 @@ func PerformSearch(query string, maxPages int, visitPages bool) []structures.Res
 	return results
 }
 
-func runEngines(toSearch []Engine, query string, worker *conc.WaitGroup, relay *structures.Relay, options *structures.Options) {
-	for _, eng := range toSearch {
-		switch Engine(eng) {
-		case Google:
-			worker.Go(func() {
-				err := google.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", google.SEDomain)
-				}
-			})
-		case DuckDuckGo:
-			worker.Go(func() {
-				err := duckduckgo.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", duckduckgo.SEDomain)
-				}
-			})
-		case Mojeek:
-			worker.Go(func() {
-				err := mojeek.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", mojeek.SEDomain)
-				}
-			})
-		case Qwant:
-			worker.Go(func() {
-				err := qwant.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", qwant.SEDomain)
-				}
-			})
-		case Etools:
-			worker.Go(func() {
-				err := etools.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", etools.SEDomain)
-				}
-			})
-		case Swisscows:
-			worker.Go(func() {
-				err := swisscows.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", swisscows.SEDomain)
-				}
-			})
-		case Brave:
-			worker.Go(func() {
-				err := brave.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", brave.SEDomain)
-				}
-			})
-		case Bing:
-			worker.Go(func() {
-				err := bing.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", bing.SEDomain)
-				}
-			})
-		case Startpage:
-			worker.Go(func() {
-				err := startpage.Search(context.Background(), query, relay, options)
-				if err != nil {
-					log.Error().Err(err).Msgf("Failed searching %v", startpage.SEDomain)
-				}
-			})
+func runEngines(engines map[structures.Engine]config.Engine, query string, worker *conc.WaitGroup, relay *structures.Relay, options *structures.SEOptions) {
+	log.Info().Msgf("Enabled engines: %v", config.EnabledEngines)
+
+	for name, engine := range engines {
+		switch name {
```

Comment:
This should be changed so iteration over EnabledEngines is possible.

Reply:
I forgot to impl that. Thanks for noticing!